src/charm.py

```diff
@@ -30,105 +21,63 @@ class CephFSServerProxyCharm(ops.CharmBase):
-            return
-
         config = {k: self.config.get(k) for k in CephFSServerProxyCharm._REQUIRED_CONFIGS}
 
         # This method catches both uninitialized configs and empty strings/lists.
         missing = [k for (k, v) in config.items() if not v]
 
         if missing:
-            msg = f"Missing config key{'s' if len(missing) > 1 else ''} {', '.join(missing)}"
+            if len(missing) > 1:
+                fmt = "Missing config keys {}"
+            else:
+                fmt = "Missing config key {}"
+
+            msg = fmt.format(", ".join(missing))
+
             logger.error(msg)
             self.unit.status = ops.BlockedStatus(msg)
             return
 
         # All configs are set from this point.
         fsid = str(config["fsid"])
-        fs_name, fs_path = str(config["sharepoint"]).split(
-            ":"
-        )  # Expected format: <filesystem_name>:<filesystem_shared_path>
-        monitor_hosts = str(
-            config["monitor-hosts"]
-        ).split()  # Expected format: <ip/hostname>:<port> <ip/hostname>:<port>
-        username, cephx_key = str(config["auth-info"]).split(
-            ":"
-        )  # Expected format: <username>:<cephx-base64-key>
-
-        # Assume all config is correct. The relation library should be the one in charge of validating inputs.
-        self.set_state(
-            "config",
-            {
-                "fsid": fsid,
-                "name": fs_name,
-                "path": fs_path,
-                "monitor_hosts": monitor_hosts,
-                "username": username,
-                "key": cephx_key,
-            },
-        )
-
-        self.unit.status = ops.ActiveStatus("Exporting share")
 
-    def _on_share_requested(self, event: ShareRequestedEvent) -> None:
-        """Handle when CephFS client requests a CephFS share."""
-        logger.debug(("CephFS share requested with parameters " f"('name'={event.name})"))
-        if not (mount_info := self.mount_info):
-            logger.warning("Deferring ShareRequested event because filesystem info is not set")
-            self.unit.status = ops.BlockedStatus("No configured filesystem info")
-            event.defer()
-            return
-        share_info, auth_info = mount_info
-        logger.debug(f"CephFS mount info set to {share_info}")
-        self._cephfs_share.set_share(event.relation.id, share_info=share_info, auth_info=auth_info)
-
-    @property
-    def mount_info(self) -> Optional[Tuple[CephFSShareInfo, CephFSAuthInfo]]:
-        """Get the required info to mount the proxied CephFS share."""
-        if not (config := self.get_state("config")):
+        # Expected format: <filesystem_name>:<filesystem_shared_path>
+        sharepoint_cfg = str(config["sharepoint"])
+        sharepoint = sharepoint_cfg.split(":", maxsplit=1)
+        if len(sharepoint) != 2 or not sharepoint[0] or not sharepoint[1]:
+            msg = f"Invalid sharepoint `{sharepoint_cfg}`"
+            logger.error(msg)
```

Comment:
[nit]: all lowercase log message here? Or will that accidentally clobber information we don't want clobbered?

```suggestion
            logger.error(msg.lower())
```

If `.lower()` would clobber the share configuration info, you could have `msg` be all lowercase and then capitalize it for the status message:

```suggestion
            msg = f"invalid sharepoint `{sharepoint_cfg}`"
            logger.error(msg)
            self.unit.status = ops.BlockedStatus(msg.capitalize())
```